appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/StatefulSessionContainer.java

```diff
@@ -1893,31 +1745,25 @@ protected void postInvokeTx(EjbInvocation inv) throws Exception {
      *
      * @return true if the removal should be skipped, false otherwise.
      */
-    private boolean retainAfterRemoveMethod(EjbInvocation inv,
-                                            EjbRemovalInfo rInfo) {
-
-        boolean retain =
-                (rInfo.getRetainIfException() &&
-                        (inv.exceptionFromBeanMethod != null) &&
-                        (isApplicationException(inv.exceptionFromBeanMethod)));
-
-        return retain;
-
+    private boolean retainAfterRemoveMethod(EjbInvocation inv, EjbRemovalInfo rInfo) {
+        return rInfo.getRetainIfException() && isApplicationException(inv.exceptionFromBeanMethod);
     }
 
     /**
      * Called from preInvoke which is called from the EJBObject for local and
      * remote invocations.
      */
+    @Override
     public void releaseContext(EjbInvocation inv) {
         SessionContextImpl sc = (SessionContextImpl) inv.context;
 
         // Make sure everything is within try block so we can be assured that
         // any instance lock is released in the finally block.
         try {
             // check if the bean was destroyed
-            if (sc.getState() == BeanState.DESTROYED)
+            if (sc.getState() == BeanState.DESTROYED) {
```

Comment:
Maybe `sc.getState()` should be extracted to a local var to be deterministic throughout the method?

Reply:
What if author made it intentionaly? The status changes in each step?

Reply:
You are right, we should keep such a change out of this PR. In this old code one never knows.